README.rst
#######################################
Python 3: From None to Machine Learning
#######################################


:Title: Python 3: from None to Machine Learning
:Author: Matt Harasymczuk
:Language: English
:ISBN: 978-83-957186-2-5
:Year: 2015-2022
:Online Access: https://python3.info
:License: Creative Commons Attribution-ShareAlike 4.0 International License


Author
======
.. figure:: _static/img/AstroMatt.jpg

.. csv-table::
    :widths: 15, 65

    "author", "`Matt Harasymczuk <https://www.astronaut.center>`_"
    "email", "matt@astrotech.io"
    "www", "https://www.astronaut.center"
    "github", "https://github.com/astromatt"
    "linkedin", "https://linkedin.com/in/mattharasymczuk"
    "facebook", "https://facebook.com/matt.harasymczuk"

.. csv-table:: Other Books from Author. Learn more at https://www.astronaut.center/books
    :widths: 25, 20, 55
    :header: "ISBN", "Online Access", "Title"

    "978-83-957186-2-5", "https://python3.info", "Python 3: from None to Machine Learning"
    "978-83-957186-3-2", "https://dev.astrotech.io", "The Software Engineering: DevOps, CI/CD, Docker, Provisioning and Git Flow"
    "978-83-957186-0-1", "https://www.astronaut.pl", "Astronaut Selection and Training for Long Duration Spaceflight and Extraterrestrial Activity"
    "978-83-956752-0-1", "https://www.astronaut.center/books", "Space in Practice: How to Prepare and Conduct Stratospheric Balloon Mission"
    "978-83-957186-4-9", "https://pl.habitatos.space", "HabitatOS - Development of operating system prototype for Lunar and Martian habitats"
    "978-83-957186-1-8", "https://alsep.astronaut.center", "Geophysics experiments from Apollo Lunar Surface Experiments Package"


Big Picture
===========
.. figure:: about/img/python-training-path.png


Note
====
* For consulting or training course requests please email matt@astrotech.io
* Before training course please setup your environment
* More information in `Install`


about/history.rst
Python History
==============
* http://python-history.blogspot.com/

Python is a high-level, general-purpose programming language.
Its design philosophy emphasizes code readability with the use
of significant indentation via the off-side rule.
[#wikiPython]_

Python is dynamically typed and garbage-collected. It supports multiple
programming paradigms, including structured (particularly procedural),
object-oriented and functional programming. It is often described as a
"batteries included" language due to its comprehensive standard library.
[#wikiPython]_


Autor
-----
Guido van Rossum (born 31 January 1956) is a Dutch programmer best known
as the creator of the Python programming language, for which he was the
"benevolent dictator for life" (BDFL) until he stepped down from the position
on 12 July 2018. He remained a member of the Python Steering Council
through 2019, and withdrew from nominations for the 2020 election.
[#wikiPython]_


Name
----
Python's name is derived from the British comedy group Monty Python,
whom Python creator Guido van Rossum enjoyed while developing the language.
Monty Python references appear frequently in Python code and culture;
for example, the metasyntactic variables often used in Python literature
are spam and eggs instead of the traditional foo and bar. The official
Python documentation also contains various references to Monty Python routines.
[#wikiPython]_


History
-------
Python was conceived in the late 1980s by Guido van Rossum at Centrum
Wiskunde & Informatica (CWI) in the Netherlands as a successor to the
ABC programming language, which was inspired by SETL, capable of exception
handling and interfacing with the Amoeba operating system. Its implementation
began in December 1989. Van Rossum shouldered sole responsibility for the
project, as the lead developer, until 12 July 2018, when he announced his
"permanent vacation" from his responsibilities as Python's "benevolent
dictator for life", a title the Python community bestowed upon him to
reflect his long-term commitment as the project's chief decision-maker.
In January 2019, active Python core developers elected a five-member
Steering Council to lead the project. [#wikiPython]_

Python 2.0 was released on 16 October 2000, with many major new features
such as list comprehensions, cycle-detecting garbage collection, reference
counting, and Unicode support. Python 3.0, released on 3 December 2008, with
many of its major features backported to Python 2.6.x and 2.7.x. Releases
of Python 3 include the 2to3 utility, which automates the translation of
Python 2 code to Python 3. [#wikiPython]_

Python 2.7's end-of-life was initially set for 2015, then postponed to
2020 out of concern that a large body of existing code could not easily
be forward-ported to Python 3. No further security patches or other
improvements will be released for it. [#wikiPython]_


References
----------
.. [#wikiPython] Wikipedia Contributors. Python (programming language). Year: 2023. Retrieved: 2023-04-14. URL: https://en.wikipedia.org/wiki/Python_(programming_language)


_static/js/menu-cite.js
let today = new Date()
let year = today.getFullYear()
let datetime = today.toISOString()
let date = datetime.split('T')[0]
let url = window.location.href
let revision = $('footer .commit code').text()
let chapter = $('div[itemprop="articleBody"] h1').text()

const CITATION = `<div id="cite" itemscope itemtype="http://schema.org/ScholarlyArticle">
    <label for="citation">Cite current page:</label>
    <blockquote itemprop="citation" id="citation">

        <!-- author -->
        <span itemprop="author" itemscope itemtype="http://schema.org/Person">
            <link itemprop="url" href="https://orcid.org/0000-0002-2961-0617" />
            <span itemprop="name"><span itemprop="familyName">Harasymczuk</span>, <span itemprop="givenName">Matt</span></span>
        </span>.

        <!-- title -->
        <cite itemprop="headline">Python: from None to Machine Learning.</cite>

        <!-- chapter -->
        <span itemscope itemtype="https://schema.org/Chapter" >
            <span itemprop="name">${chapter}</span>
             <link itemprop="isPartOf" itemscope itemtype="https://schema.org/Book" itemid="https://python3.info" />
        <span>

        <!-- year -->
        (<time itemprop="datePublished" datetime="${datetime}">${year}</time>)

        <!-- publisher -->
        <span itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
            <span itemprop="name">Analog Astronaut Training Center</span>.
        </span>

        <!-- version -->
        ISBN: <span itemprop="isbn">978-83-957186-2-5</span>.
        Revision: <span itemprop="version">${revision}</span>.

        <!-- url -->
        Retrived: <time itemprop="dateModified" datetime="${datetime}">${date}</time>.
        URL: <link itemprop="url" href="${url}">${url}</link>

        <!-- license -->
        <!-- License: <link itemprop="license" rel="license" href="https://creativecommons.org/licenses/by-sa/4.0/">CC-BY-SA-4.0</link>-->
    </blockquote>`;


document.addEventListener("DOMContentLoaded", () => {
    let where = $('footer');
    where.append(CITATION);
});
